### DISCUSSION TITLE
PERMANENTLY CONNECT

### DISCUSSION THREAD

**Discussion by Alekie74** [kakzaki/blue_thermal_printer#161] [General]
how can I tie the connected printer so that no need to connect every time the application is loaded.

**Comment by calimba** [comment]
Try this
```dart
final myDevice = BluetoothDevice.fromMap({
      'name': data['name'],
      'address': data['address'],
      'type': data['type']
});
```

**Comment by Alekie74** [comment]
where do I put this code?

**Comment by Alekie74** [comment]
Hi Calimba,

Kindly guide on where I should insert this piece of code for the bluetooth
printer to be permanently remembered whenever the printing is done.

**Comment by calimba** [comment]
I do this
```dart
void _checkIfLoggedIn() async {
      SharedPreferences localStorage = await SharedPreferences.getInstance();
      var token = localStorage.getString('token');
      var userJson = localStorage.getString('user'); 
      var user = json.decode(userJson);
      var stringValue = user['dispositivo'].toString();
      var stringValue2 = user['address'].toString();
      var stringValue3 = user['tipo'];
      if(token != null){
        setState(() => {
            _isLoggedIn = true,
            userData = user,
            _dispositivo = stringValue,
            _direccion = stringValue2,
            _tipo = stringValue3
        });
      }

      if(_direccion != null){

          final myDevice = BluetoothDevice.fromMap({
            'name': _dispositivo,
            'address': _direccion,
            'type': _tipo
        });

        bluetooth.isConnected.then((isConnected) {
          
            if (!isConnected) {
              
              bluetooth.connect(myDevice).catchError((error) {
                setState(() {
                  _connected = false;
                  });
              });
            }
          });
      }
  }
```

**Comment by ebrahimmansur** [comment]
i used the root widget with the widget life cycle  mixing to invoke the search and connect and disconnect  with the bluetooth device

**Comment by Alekie74** [comment]
Kindly expound.

Regards,
Alex Kithome.

**Comment by ebrahimmansur** [comment]
Here is the approach.....
- make a statefull widget with a mixin widget binding observe  and call it for example app Life cycle and implement the function

- wrap the root screen of your app with the app life cycle widget 

- inside the app cycles widget will hold a list of services that need to start or stop based on the app life cycle...
Make an interface that has a 
     - onAppStart 
     --onAppClosed 
And have all the services implement them ...and based on the life cycle state invoke the onStart or onclosed

....if you need more explanation here is my
LinkedIn handler 

https://www.linkedin.com/in/ebrahim-mansur-51b526202
